pkg/operation/botanist/etcd.go

```diff
@@ -121,7 +121,10 @@ func (b *Botanist) DeployEtcd(ctx context.Context) error {
 			LeaderElection:       backupLeaderElection,
 		})
 
-		if gardencorev1beta1helper.SeedSettingOwnerChecksEnabled(b.Seed.GetInfo().Spec.Settings) {
+		// Owner checks are only enabled if the `etcd-main` resource is deployed with 1 replica.
+		// They must not be used for clustered etcd. Ref: https://github.com/gardener/gardener/issues/6302
+		if gardencorev1beta1helper.SeedSettingOwnerChecksEnabled(b.Seed.GetInfo().Spec.Settings) &&
+			getReplicas(b.Shoot.GetInfo()) == 1 {
```

Comment:
Hi @plkokanov,
Thanks for PR. Have you tested the scale-up scenario for OwnerCheck? 
Basically just wanted to check when etcd replica=1 (when ownerCheck is on) scale-up to etcd replica=3. Then ownerCheck suppose to be get turned off for all 3 replicas.

Reply:
Nope, I did not test it because currently there is a validation plugin that forbids adding the `alpha.control-plane.shoot.gardener.cloud/high-availability` annotation if the shoot was not created with it directly. So I was not sure if I should test that case currently.
But I could give it a go by temporarily disabling the plugin or playing with enabling/disabling the feature gate.

Reply:
This works because the backup struct for `e.etcd.Spec.Backup` is freshly assigned and calculated in every reconcile loop: https://github.com/gardener/gardener/blob/2391302ccf38eb172b1a035f9d4da433222e0e65/pkg/operation/botanist/component/etcd/etcd.go#L529